### DISCUSSION TITLE
Installer project

### DISCUSSION THREAD

**Discussion by Zilkode** [AvaloniaUI/Avalonia#6088] [General]
Hi all.
So i came across the situation where i want the end user to install my software (downloaded as single executable).
Because i couldn't find a way to use "installer project" with Avalonia i was trying to think a way to do so.
my thought was:
1. Publish my software to an Output folder and compress it.
2. Create another Avalonia project which will be my installation project.
3. Add the software zip file as an asset to the custom made installation project.
4. Publish the install project as single executable.
5. Once downloaded and executed, the installer do whatever i want.

Before doing so i had a reject of my own - there will be a lot of duplicate files (.net core and Avalonia libraries, once in the software and another in the installer) in this mongrel project which will result a bloated executable file.

Of course there is a way to publish both as platform dependant (to exclude .net core libraries completely)  but i am trying to avoid "must haves" installations.

As for now i am bushwhacking to find a way exracting core libraries from the installer project into the "installation path", that way i will be able to exclude those from the "software project" - no much success so far.

Has anyway ever faced this situation or have an idea how to accomplish a reasonable solution?

**Comment by kekekeks** [comment]
For installers the recommended approach is to use platform-specific options:
- Windows: use NSIS (really simple to setup) or WIX Toolset (more complex setup, but generates native MSI)
- Linux: use https://github.com/qmfrederik/dotnet-packaging to generate deb/rpm packages and ship tar.gz for the rest
- macOS: follow the guide here https://docs.avaloniaui.net/docs/distribution-publishing/macos

**Comment by maxkatz6** [comment]
> Because i couldn't find a way to use "installer project" with Avalonia i was trying to think a way to do so.

Avalonia does not forces you to use some specific way to install the application. Neither it implements it without you. It's GUI framework, not packaging system. Search for dotnet packaging and publishing.
Apart from what @kekekeks already said, I would recommend MSIX on Windows 10, if it's applicable in your scenario.

**Comment by Zilkode** [comment]
Thanks for your suggestions, i'll explore them.
